cmd/cat/cat.go

```diff
@@ -0,0 +1,49 @@
+// Based on Plan9 cat
+
+package main
+
+import (
+	"log"
+	"os"
+)
+
+func cat(f *os.File, name string) {
+	var (
+		buf [8192]byte
+	)
+
+	for {
+		n, err := f.Read(buf[:])
+
+		if err != nil {
+			log.Fatal("error reading %s: %s", name, err)
```

Comment:
You're right about the EOF too. I coded without reading the online docs, I hate this behavior of returning error when there's no error at all.